addons/sale_project/models/project.py

```diff
@@ -823,22 +821,26 @@ def _group_expand_sales_order(self, sales_orders, domain, order):
             return search_on_comodel
         return sales_orders
 
-    @api.depends('sale_line_id', 'project_id', 'partner_id.commercial_partner_id', 'allow_billable')
+    @api.depends('sale_line_id', 'project_id.sale_order_id', 'partner_id.commercial_partner_id', 'allow_billable')
     def _compute_sale_order_id(self):
         for task in self:
             if not task.allow_billable:
                 task.sale_order_id = False
                 continue
-            sale_order_id = task.sale_order_id or self.env["sale.order"]
-            if task.sale_line_id:
-                sale_order_id = task.sale_line_id.sudo().order_id
-            elif task.project_id.sale_order_id:
-                sale_order_id = task.project_id.sale_order_id
-            if task.partner_id.commercial_partner_id != sale_order_id.partner_id.commercial_partner_id:
-                sale_order_id = False
-            if sale_order_id and not task.partner_id:
-                task.partner_id = sale_order_id.partner_id
-            task.sale_order_id = sale_order_id
+            sale_order = (
+                task.sale_line_id.order_id
```

Comment:
is the `sale_line_id.sudo()` no longer necessary? it was added in this bugfix 8a57e84a320f3ea8d867aa850d357a89c91af09a, since there is was no added test to that, I don't know if removing the sudo will cause a regression

Reply:
That `sudo()` wasn't even necessary in that commit. The bug existed in v12.0, but the fix didn't have to be forward ported to 13.0 or later.

Not sure how `compute_sudo` was supposed to work back before `env.su` was a thing, but for whatever reason, you weren't given superuser privileges entering that compute method in v12.0, despite `compute_sudo` being enabled, which is why the `sudo()` was added.

Starting from v13.0, with `env.su` being introduced, the compute method does work as expected, even without the extra `sudo()`.

Reply:
Could be needed if the project user changes the customer and he does not have any access to Sales app, I think (I did not check). 🙂

Reply:
Added this to the test just to be sure, changing a task's `partner_id` while not having access to Sales doesn't affect the compute method: https://github.com/odoo/odoo/blob/73075f3046621fd645691bd7b84ab498cf0ec5df/addons/sale_project/tests/test_sale_project.py#L813-L816